linkml/generators/common/name_compatibility.py

```diff
@@ -0,0 +1,84 @@
+        """Specify the naming policy on instantiation"""
+        self.profile = profile
+        self.do_not_fix = do_not_fix
+
+    def _no_heading_digits(self, input: str) -> str:
+        """Ensure name does not start with a heading digit"""
+        output = input
+        if self.re_heading_digit.match(input):
+            if self.do_not_fix:
+                raise ValueError(f"Name '{input}' starts with digit (illegal GraphQL) and will not be fixed!")
+            output = f"__{input}"
+        return output
+
+    def _transform_char(self, char: str) -> str:
+        """Transform unsupported characters"""
+        if len(char) != 1:
+            raise Exception(f"Single character expected, but got '{char}'!!")
+        # try to use names for ASCII characters
+        if ord(char) < 128:
+            try:
+                # unicodedata.lookup should be able to invert the transformation
+                return f"__{unicodedata.name(char).replace(' ','_')}__"
+            except ValueError:
+                pass
+        # fallback to code-transformation if none of the previous has worked
+        return f"__U_{ord(char):04X}_"
+
+    def _only_alphanum_underscore(self, input: str) -> str:
+        """Ensure name does not contain any unsupported characters"""
+        output = input
+        # with re.split and re.findall we get in the same order and separated in two arrays
+        # the substrings between special characters and the special characters
+        no_alphanum_underscore_match = self.re_no_alphanum_underscore.findall(input)
+        if no_alphanum_underscore_match:
+            if self.do_not_fix:
+                raise ValueError(f"Name '{input}' contains a character illegal in GraphQL and will not be fixed!")
+            to_keep = self.re_no_alphanum_underscore.split(input)
+            # first comes first substring to keep
+            output = to_keep[0]
+            # each char replacement is followed by the next substring to keep
+            for offset in range(0, len(to_keep) - 1):
+                output = output + self._transform_char(no_alphanum_underscore_match[offset])
+                output = output + to_keep[offset + 1]
+        return output
+
+    def _graphql_compatibility(self, input: str) -> str:
+        """Ensure name compatibility with GraphQL name policies"""
+        # as of now, some (hopefully) very rare patterns are reserved to mark transformations
+        if self.re_reserved_heading_digit.match(input):
+            raise NotImplementedError("Names starting with a double underscore followed by a digit are not supported!!")
+        if self.re_reserved_unicode_name_transformation.match(input):
+            raise NotImplementedError("Names containing valid names between double underscores are not supported!!")
+        if self.re_reserved_unicode_code_transformation.match(input):
+            raise NotImplementedError("Names containing strings like '__U_xxxx_' are not supported!!")
```

Comment:
The test cases are already in place. I prefer to be careful blocking something very improbable for the time being. The idea of the name inverter appears in different places, but without any clear plan. So I am leaving it as it is now. 

It could be better, but as this PR is now, it is already much better than what we gen-graphql provides now 😄